Source/Core/Core/AchievementManager.cpp

```diff
@@ -812,11 +812,15 @@ void AchievementManager::HandleAchievementChallengeIndicatorHideEvent(
 void AchievementManager::HandleAchievementProgressIndicatorShowEvent(
     const rc_client_event_t* client_event)
 {
-  const auto& instance = AchievementManager::GetInstance();
+  auto& instance = AchievementManager::GetInstance();
+  auto current_time = std::chrono::steady_clock::now();
+  if (current_time - instance.m_last_progress_message < std::chrono::seconds{2})
```

Comment:
Maybe unnecessary complexity but would it make sense to build the timeout on top of the [fadeout time](https://github.com/dolphin-emu/dolphin/blob/b0f8520a3d8fc39bbd4364f9c382b97363566936/Source/Core/VideoCommon/OnScreenDisplay.cpp#L31)?

Something like:

```
const auto message_wait_time = std::chrono::millisecond{OSD::MESSAGE_FADE_TIME}*2;
if (current_time - instance.m_last_progress_message < message_wait_time )
```

Reply:
Fair, probably should have thought of that.